Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -223,15 +235,17 @@ extension PackageGraph {
                 let pluginOutputDir = outputDir.appending(components: package.identity.description, target.name, pluginTarget.name)
 
                 // Invoke the plugin.
-                let result = try pluginTarget.invoke(
+                let result = try tsc_await { pluginTarget.invoke(
                     action: .createBuildToolCommands(target: target),
                     package: package,
                     buildEnvironment: buildEnvironment,
                     scriptRunner: pluginScriptRunner,
                     outputDirectory: pluginOutputDir,
                     toolNamesToPaths: toolNamesToPaths,
+                    fileSystem: fileSystem,
                     observabilityScope: observabilityScope,
-                    fileSystem: fileSystem)
+                    on: DispatchQueue(label: "plugin-invocation"),
```

Comment:
.sharedConcurrent ?

Reply:
It doesn't sound as if that will provide the ordering of output data from the plugin that we'd want.